### DISCUSSION TITLE
How to keep metrics using internal DNS

### DISCUSSION THREAD

**Discussion by gabrielmcf** [linkerd/linkerd2#6089] [Q&A]
Hi everyone!
We use in our architecture an Nginx ingress, together with Linkerd. We've started to use internal DNS pointing to an internal ingress, in order to organize the network. The problem is that we loose the visibility of which pod is hitting which one, because all the traffic is proxied through the ingress.
Is there a way to keep this visibility while using the internal ingress?

**Comment by adleong** [comment]
Since your pods are no longer talking to each other directly, you would need something like distributed tracing to see how the requests flow through the ingress.  Take a look at https://linkerd.io/2.10/tasks/distributed-tracing/ to see how this can be done with Linkerd.

**Reply by gabrielmcf** [reply]
Isn't there any annotation I could use in my ingress to pass through the right headers on this routes?
